src/main.rs

```diff
@@ -123,11 +123,30 @@ impl State {
     /// Initialize the window and its EGL surface.
     fn init_window(&mut self, connection: &mut Connection, queue: &QueueHandle<Self>) {
         // Initialize EGL context.
-        let native_display = NativeDisplay::new(connection.display());
-        let display = EGLDisplay::new(&native_display, None).expect("Unable to create EGL display");
-        let context =
-            EGLContext::new_with_config(&display, GL_ATTRIBUTES, Default::default(), None)
-                .expect("Unable to create EGL context");
+        let mut raw_display = WaylandDisplayHandle::empty();
+        raw_display.display = connection.backend().display_ptr().cast();
+        let raw_display = RawDisplayHandle::Wayland(raw_display);
+
+        let display =
+            unsafe { Display::from_raw(raw_display).expect("Unable to create EGL display") };
+
+        let config_template = ConfigTemplateBuilder::new().with_api(Api::GLES2).build();
+        let config = unsafe {
+            display
+                .find_configs(config_template)
+                .ok()
+                .and_then(|mut configs| configs.next())
+                .expect("No suitable configuration found")
+        };
```

Comment:
Why is this unsafe? As far as I can tell from the Rust side there's not much you could be doing wrong? It's not like `config_template` is a pointer.

Reply:
Because you can pass raw pointer in `config_template` to native window to build it in a most compatible way with present native window.

Reply:
And that pointer cannot be checked in the function where you pass it?

Reply:
Yeah, they don't.